src/app/theme/SplashScreenService.ts

```diff
@@ -72,28 +78,155 @@ export const showSplashScreen = (
+      `Auto-hide duration reached for splash screen "${bootsplashName}"`
+    )
+
+    logToSentry(
+      new Error(
+        `Splashscreen reached autoHideDuration with bootsplashName "${bootsplashName}"`
+      )
+    )
+
+    void manageTimersAndHideSplashScreen(bootsplashName, true)
+  }, autoHideDuration)
+
+  splashScreenLogger.debug(
+    `Setting timeout with ID "${JSON.stringify(
+      autoHideTimers[bootsplashName]
+    )}" for splash screen "${bootsplashName}"`
+  )
+}
+
+const manageTimersAndHideSplashScreen = async (
+  bootsplashName: SplashScreenEnum | undefined = splashScreens.GLOBAL,
+  fromTimeout = false
+): Promise<void> => {
+  if (bootsplashName !== splashScreens.SECURE_BACKGROUND)
+    destroyTimer(bootsplashName, fromTimeout)
+
+  try {
+    await RNBootSplash.hide({ fade: true, bootsplashName })
+  } catch (error) {
+    splashScreenLogger.error(
+      `Error managing timers and hiding splash screen "${bootsplashName}"`,
+      error
+    )
+    logToSentry(error)
+  }
+}
+
+const destroyTimer = (
+  bootsplashName: SplashScreenEnum,
+  fromTimeout = false
+): void => {
+  const timer = autoHideTimers[bootsplashName]
+
+  if (autoHideTimers[bootsplashName]) {
+    if (fromTimeout) {
+      splashScreenLogger.debug(
+        `Destroying existing timer with ID "${JSON.stringify(
+          timer
+        )}" for splash screen "${bootsplashName}" after auto-hide duration reached`
+      )
+    } else {
+      splashScreenLogger.debug(
+        `Clearing existing timer with ID "${JSON.stringify(
+          timer
+        )}" for splash screen "${bootsplashName}" after manual hide`
+      )
+    }
+
+    clearTimeout(autoHideTimers[bootsplashName])
+    Reflect.deleteProperty(autoHideTimers, bootsplashName)
+  }
 }
+
+let activeTimersAtBackground: Record<string, boolean> = {}
+
+const resetTimersOnActive = (): void => {
+  splashScreenLogger.debug(
+    `App is becoming active. Evaluating timers to reset...`
+  )
+
+  const timersToReset = Object.keys(activeTimersAtBackground)
+
+  if (timersToReset.length === 0) {
+    splashScreenLogger.info(`No splash screen timers to reset.`)
+  } else {
+    timersToReset.forEach(splashScreen => {
+      if (activeTimersAtBackground[splashScreen]) {
+        splashScreenLogger.info(
+          `Resetting timer for splash screen "${splashScreen}"`
+        )
+        setTimeoutForSplashScreen(splashScreen as SplashScreenEnum)
+      }
+    })
+  }
+
+  // Clear the record once timers are reset
+  activeTimersAtBackground = {}
+}
+
+const clearTimersOnBackground = (): void => {
+  splashScreenLogger.debug(
+    `App is going to background. Clearing active timers...`
+  )
+
+  Object.keys(autoHideTimers).forEach(splashScreen => {
+    if (autoHideTimers[splashScreen]) {
+      splashScreenLogger.info(
+        `Clearing timer for splash screen "${splashScreen}". Will reset on active.`
+      )
+      activeTimersAtBackground[splashScreen] = true // Mark as active
```

Comment:
nit: Seems like values is always `true` some maybe we can use an array instead of a dict? So we would not have to use the `Object.keys` stuff on line 190